spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ehcache/EhCacheProperties.java

```diff
@@ -0,0 +1,57 @@
+package org.springframework.boot.autoconfigure.ehcache;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * Configuration properties to configure {@link org.springframework.cache.ehcache.EhCacheFactoryBean}.
+ *
+ * @author Eddú Meléndez
+ * @since 1.3.0
+ */
+@ConfigurationProperties(prefix = "ehcache", ignoreUnknownFields = false)
+public class EhCacheProperties {
+
+    @NotNull
+    private String location = "ehcache.xml";
+
+    private String name;
+
+    private boolean checkLocation = false;
+
+    private boolean shared = false;
```

Comment:
You need to add field javadoc on these 4 fiels so that those show up in the meta-data. Look at other `@ConfigurationProperties` for  some examples.